### ISSUE TITLE
Bug: [no-inferrable-types]  False negatives on auto-accessors

### ISSUE THREAD

**Issue by yeonjuan** [typescript-eslint/typescript-eslint#10763]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.7.2&fileType=.ts&code=FAYwNghgzlAEBiB7RsDexawA4CdFagC5YA7AVwFsAjAUx1gF5YBWAbg1ghBBpkXq48%2BOAAp4CxctTqMW7AL5A&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1uYDNFpoAQwBGSWkVKoMkftA7RI4MAF8QyoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
class Foo {
  props: number = 5; // Error (ok)
  accessor accessorProps: number = 5; // No Error (false negative)
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
     "@typescript-eslint/no-inferrable-types": "error"
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result
```ts
class Foo {
  props: number = 5; // Error
  accessor accessorProps: number = 5; // Error
}
```

### Actual Result
```ts
class Foo {
  props: number = 5; // Error
  accessor accessorProps: number = 5; // No Error
}
```

**Comment by ronami** [comment]
This seems like a bug to me, as the type remains the same with or without the annotation ([playground link](https://typescript-eslint.io/play/#ts=5.7.2&showAST=es&fileType=.ts&code=FAYwNghgzlAEBiB7RsDexawiEBTGiATljvlEQAqGIAOUAjAFywB2ArgLYBGuxAvLACsAbgywA9OMzTMAPQD8wMdjwFiKspWp0ATLAEixkmdIXAAvkA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1uYDNFpoAQwBGSWkVKoMkftA7RI4MAF8QyoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)).

**Comment by ronami** [comment]
I was thinking about this some more: I think that missing `accessor` handling seems pretty common because it's a different AST node rather than an attribute (like `readonly`). I did a quick search and I think I found six additional bugs with six different rules:

- `explicit-module-boundary-types` ([playground link](https://typescript-eslint.io/play/#ts=5.7.2&fileType=.ts&code=KYDwDg9gTgLgBAYwDYEMDOa4BU4G8BQccA9MXFMJLMACaLQUIxICehcAZhBHALxwAKAJR8AfHjgBffO1JwAdhHgUqMWnABGAV3hoAFhC1I6G4OxQIEwDNE0oofQSN7jcU-NKA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6RAD2MbMroFsA9gBMEiWgCNBsJsICG0QrSKlUGSImjRB0SODABfEAaA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false))
- `prefer-return-this-type` ([playground link](https://typescript-eslint.io/play/#ts=5.7.2&fileType=.ts&code=MYGwhgzhAEAq0G8BQ1oHo3QE4FMAOA9lgC44Am0wRuwxIAnitAGbQC80AFAJQBcc7AHzRiACwCWEANxImGaADsCxbPiKkKAIwCuKiKILaQWnEzDBgOKEWgBzdlz4C2wsZJkBfJEA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0Wls-yxoTWvgAWlZKJKJ0UXtAD20SODABfEOqA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false))
- `no-unsafe-assignment` ([playground link](https://typescript-eslint.io/play/#ts=5.7.2&fileType=.ts&code=MYGwhgzhAEBiD29oG8BQ1oHpPQE4FMAHeXAF3wBNpgSDhSQBPdaMaAXmgEZWYwA7RgG5ULbNH7xSeIiXJUARgFdpEABbwlIRfhZhgwfFBLQFHbr1aCRAX1RA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tieQEMAZolp9kySgHMmAW0Qt0URNGgdokcGAC%2BITUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false))
- `explicit-member-accessibility` ([playground link](https://typescript-eslint.io/play/#ts=5.7.2&fileType=.ts&code=MYGwhgzhAEAq0G8BQ1oHo3QE4FMAOA9lgC44Am0wRuwxIAnitGAHYCWAtmCAHJgc5oAXmgBGANxImGaCwLFs%2BIqQoAjAK4KIACwLqQanEzDBgOKEWgAzAgWFjJAXyRA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6RAD2MbMroFtE-AEaJotAIZkyKZJWGVGRdFDHQA9tEjgwAXxC6gA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false))
- `consistent-generic-constructors` ([playground link](https://typescript-eslint.io/play/#ts=5.7.2&fileType=.ts&code=MYGwhgzhAEAq0G8BQ1oHo3QE4FMAOA9lgC44Am0wRuwxIAnitKRMQIwBc0AsmHgDyssASwB2AcwA00UQFcAtgCMcWAHzQAvDJwB3HnwAUASgDcSJhhkFi2fEVIVFsmxAAWBWSEc4mYYMBwoImZA4gAmLl4BITEpGQVlNU1tPSjjMwBfJCA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6MgeyeUuX0Ra0A5gMRUytTt3xwy%2BDtFQYA2uGw4x0BZAA0a9ZCl9Z86JH1gAumoC%2BIG0A&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false))
- `class-methods-use-this` ([playground link](https://typescript-eslint.io/play/#ts=5.7.2&fileType=.ts&code=MYGwhgzhAEAq0G8BQ1oHo3QE4FMAOA9lgC44Am0wRuwxIAnitKRMQIzQC80AFAJRcAfIiaoMTAL4BuJEwzQAdgWLZ8RUhQBGAVxUQAFgW0gtOJmGDAcUIs2vEATF14DOw5KjFpJMiUiA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQMbwENlkBaAW0QBcALAewBNTZlESaBLVDSWgM18gAacNkgABSgE8ADilzR20yiRTx2AO0oB6fEVIUaDJizbVO6MAG0R2KImjRa0ITexZbo9gHN1TxAGFCYhQAFWoCSgBJMmkkCk0AQXVIzXteAlxEC0o4RFcIAF9XAF0RIoKgA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false))

I'll open an issue for each of these 🙃
